NewsApp/NewsApp/Model/Network/GetTopHeadlines.swift

```diff
@@ -16,16 +16,19 @@ struct GetTopHeadlines: APIRequest {
     }
 
     private let country: Country?
-    private let category: Category?
+    private let category: CategoryType?
     private let query: String?
     private let apiKey = "2c64fe5d063645f58a5cd563308d0e7c"
 
     init(
-        country: Country? = Country.fr,
-        category: Category? = Category.general,
+        category: CategoryType? = .general,
         query: String? = "") {
-        self.country = country
         self.category = category
         self.query = query
+        if let country = UserDefaults().string(forKey: Constantes.UserDefaultsKey.country) {
```

Comment:
la déclaration de API Top Headlines que représente cette classe doit être la plus simple possible et ce n'est pas à elle de faire le test de savoir si une valeur est en persistant, c'est plutôt le role du NewsService.

Reply:
fix in "FixPR"